### PR TITLE
ci: actually enable golangci-lint

### LINKED ISSUE #6320: CI isn't failing for go tests that don't compile
See: https://github.com/ipfs/go-ipfs/pull/6319

Any ideas?

### PR DESCRIPTION
**Pull request by Stebalien**
also fixes #6320 (golangci-lint _will_ fail if something doesn't compile)

### CODE REVIEW

File: core/commands/swarm.go [hunk 1]
```diff
@@ -599,8 +600,8 @@ Filters default to those specified under the "Swarm.AddrFilters" config key.
 		}
 
 		var output []string
-		for _, f := range swrm.Filters.Filters() {
-			s, err := mafilter.ConvertIPNet(f)
+		for _, f := range swrm.Filters.FiltersForAction(mafilter.ActionDeny) {
```
Comment (Stebalien) [inline core/commands/swarm.go]: Moving away from deprecated APIs.

File: plugin/plugins/gen_main.sh [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 	$shortpkg "$pkg"
 )
 
-var Plugins = $shortpkg.Plugins
+var Plugins = $shortpkg.Plugins //nolint
```
Comment (Stebalien) [inline plugin/plugins/gen_main.sh]: Don't lint because this is "unused".

File: namesys/resolve/pathresolver_test.go [hunk 3]
```diff
@@ -16,17 +16,17 @@ func TestResolveNoComponents(t *testing.T) {
 	}
 
 	_, err = resolve.Resolve(n.Context(), n.Namesys, n.Resolver, path.Path("/ipns/"))
-	if err != path.ErrNoComponents {
-		t.Fatal("Should error with no components (/ipns/).", err)
+	if err == nil {
```
Comment (magik6k) [inline namesys/resolve/pathresolver_test.go]: We probably want to check more than 'there was an error'

Comment (michaelavila) [inline namesys/resolve/pathresolver_test.go]: Probably not a big deal, but I'm curious, can we expect a specific error here and ultimately fail if we don't find the specific errors we're looking for?

Comment (michaelavila) [reply namesys/resolve/pathresolver_test.go]: It appears @magik6k and I reviewed at the same time. We are leaving the same feedback here and so I'm going to resolve this particular conversation so it's not in two threads.

Comment (Stebalien) [reply namesys/resolve/pathresolver_test.go]: But I'm lazy... fine.

**Pull request merged**